gems/sorbet-runtime/lib/types/private/methods/_methods.rb

```diff
@@ -160,13 +148,24 @@ def self._check_final_ancestors(target, target_ancestors, source_method_names)
     @was_ever_final_names.include?(method_name)
   end
 
-  def self.add_module_with_final(mod)
-    @modules_with_final.add(mod)
-    @modules_with_final.add(mod.singleton_class)
+  def self.add_module_with_final_method(mod, method_name, is_singleton_method)
+    m = is_singleton_method ? mod.singleton_class : mod
+    methods = @modules_with_final[m]
+    if methods.nil?
+      methods = Set.new
+      @modules_with_final[m] = methods
+    end
+    methods.add(method_name)
+  end
+
+  def self.note_module_deals_with_final(mod)
+    # Side-effectfully initialize the value if it's not already there
+    @modules_with_final[mod]
+    @modules_with_final[mod.singleton_class]
   end
 
   private_class_method def self.module_with_final?(mod)
-    @modules_with_final.include?(mod)
+    @modules_with_final.has_key?(mod)
```

Comment:
Why change `include?` to `has_key?` here?

Reply:
I thought I undid this, but apparently not.  I'll change it back.